irispy/iris_tools.py

```diff
@@ -672,16 +890,38 @@ def calculate_photons_per_sec_to_radiance_factor(
         for input wavelengths.
 
     """
+    # Get best fit by the method of least squares
+    iris_fit = fit_iris_xput(time_obs, time_cal_coeffs, cal_coeffs)
     # Get effective area and interpolate to observed wavelength grid.
-    eff_area_interp = _get_interpolated_effective_area(detector_type, wavelength)
+    eff_area_interp = _get_interpolated_effective_area(iris_fit, detector_type, wavelength)
     # Return radiometric conversed data assuming input data is in units of photons/s.
-    return constants.h * constants.c / wavelength / u.photon / \
-           spectral_dispersion_per_pixel / eff_area_interp / solid_angle
+    return constants.h * constants.c / wavelength / u.photon / spectral_dispersion_per_pixel / eff_area_interp / solid_angle
+
 
-def _get_interpolated_effective_area(detector_type, obs_wavelength):
-    # Get effective area
-    ########### This needs to be generalized to the time of OBS once that functionality is written #########
-    iris_response = get_iris_response(pre_launch=True)
+def _get_interpolated_effective_area(detector_type, obs_wavelength, *args, **kwargs):
```

Comment:
@kakirastern  you should have iris_fit here as an argument

Reply:
Done